pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript.md

```diff
@@ -3,120 +3,141 @@ $title: Use custom JavaScript in AMP pages
+
+`amp-bind` is generally more straightforward for basic interactions, where its tight integration into HTML tags is compelling. In this example, pressing a button changes a text fragment. AMP's data binding makes this straightforward and easy to read:
+
+```html
+<p [text]="name">Rajesh</p>
+<button on="tap:AMP.setState({name: 'Priya'})">I am Priya</button>
+```
+
+Similarly, when using an API whose output you control, you may be able to implement business logic on the server, and you may be able to format the data it outputs so that it fits smoothly into an `amp-mustache` template. `amp-list` is a good fit in such cases.
+
+`amp-bind` also provides a straightforward mechanism to communicate between AMP components. In this example, tapping on an image in an `<amp-selector>` sets the state variable `selectedSlide` to `0`, which in turn makes an `<amp-carousel>` move to its first slide.
 
 ```html
-<body>  
-  <p>Hello!</p>
-  <div>
-    <amp-script layout="container" src="customjs.js">
-    </amp-script>
-  </div>
-</body>
+<amp-carousel slide="selectedSlide">
+...
+</amp-carousel>
+
+<amp-selector>
+  <amp-img on="tap:AMP.setState({selectedSlide: 0})"/>
+</amp-selector>
 ```
 
-It would result in this:
+Traditional interactive AMP components may also be more suitable for interactions that span large sections of a webpage - as you may not wish to wrap so much of the DOM in an `<amp-script>`. Generally speaking, `amp-list` and `amp-bind` are tightly integrated with the rest of AMP, and it's straightforward to sprinkle these throughout a webpage.
+
+On the other hand, on pages that involve more complex state variables or multiple interactions, `amp-script` will likely be simpler and more maintainable. Take this example, from the [AMP Camp e-commerce demo site](https://camp.samples.amp.dev/product-details?categoryId=women-shirts&productId=79121):
 
 ```html
-<body>  
-  <p>Hello!</p>
-  <div>
-    <amp-script layout="container" src="customjs.js">
-      <span></span>
-    </amp-script>
-  </div>
-</body>
+<amp-selector
+  name="color"
+  layout="container"
+  [selected]="product.selectedColor"
+  on="select:AMP.setState({
+      product: 
+        {
+          selectedSlide: product[event.targetOption].option - 1,
+          selectedColor: event.targetOption,
+          selectedSize: product[event.targetOption].sizes[product.selectedSize] != null ?
+                        product.selectedSize : 
+                        product[event.targetOption].defaultSize,
+          selectedQuantity: 1
+        }
+      })"
+></amp-selector>
 ```
 
-## Event triggers
+This site was created before `amp-script` was released. But this sort of logic might easily be easier to write and debug in JavaScript. In general, for pages with more business logic, `amp-script` may well be less confusing and will allow you to follow better programming practices.
```

Comment:
Rewrite to cut out "easily be easier"

Reply:
There's a lot of use of "general" and "may" which seems like we're not really providing much guidance. I'd rewrite it with a stronger tone.